### PR TITLE
[AIRFLOW-3522] Add support for sending Slack attachments

### PR DESCRIPTION
**Pull request by mholtzscher**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/AIRFLOW-3522) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3522
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Adding support for sending Slack attachments with messages via the SlackWebhookOperator. This will allow customized messages to be sent with interactive content. 

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
  - Modified test_slack_webhook_hook.py
  - Modified test_slack_webhook_operator.py

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/operators/slack_webhook_operator.py [hunk 1]
```diff
@@ -55,6 +57,7 @@ def __init__(self,
                  http_conn_id=None,
                  webhook_token=None,
                  message="",
+                 attachments="",
```
Comment (feluelle) [inline airflow/contrib/operators/slack_webhook_operator.py]: Shouldn't have `attachments` a default value of type list? Probably `None`

**Review by feluelle — COMMENTED** [review]
I really like this feature :)

File: airflow/contrib/hooks/slack_webhook_hook.py [hunk 2]
```diff
@@ -38,6 +38,8 @@ class SlackWebhookHook(HttpHook):
     :type webhook_token: str
     :param message: The message you want to send on Slack
     :type message: str
+    :param attachments: The attachments you want to send on Slack
+    :type attachments: list
```
Comment (rmn36) [inline airflow/contrib/hooks/slack_webhook_hook.py:43]: Can you be more specific about what this is a list of? List of strings, list of dict, list of? And what those objects represent? I'm assuming it's a list of strings that are filenames you want to attach but that should be explicitly stated

Comment (mholtzscher) [reply airflow/contrib/hooks/slack_webhook_hook.py:43]: Slack kind of misuses the work attachments in my opinion. It is a list of dicts and not really files that you are attaching. I will update the doc to be clearer

File: airflow/contrib/operators/slack_webhook_operator.py (hunk 1 above)
Comment (mholtzscher) [reply airflow/contrib/operators/slack_webhook_operator.py]: i think a default of None here makes the most sense. I will update

File: airflow/contrib/operators/slack_webhook_operator.py [hunk 3]
```diff
@@ -37,6 +37,8 @@ class SlackWebhookOperator(SimpleHttpOperator):
     :type webhook_token: str
     :param message: The message you want to send on Slack
     :type message: str
+    :param attachments: The attachments you want to send on Slack
+    :type attachments: list
```
Comment (phoenix24) [inline airflow/contrib/operators/slack_webhook_operator.py:42]: in the similar spirit as above code-review, this doc string could also be a made a little verbose.

Comment (mholtzscher) [reply airflow/contrib/operators/slack_webhook_operator.py:42]: Updated.

**Comment by mholtzscher** [comment]
@feluelle  @rmn36 @phoenix24 are any of you able to merge this?

**Comment by feluelle** [comment]
No, but maybe @ashb can take a look.

**Pull request merged**